akka-remote/src/test/scala/akka/remote/CryptoServiceProviderSpec.scala

```diff
@@ -0,0 +1,149 @@
+
+  private val knownFilesConfig = fileConfigString.withFallback(referenceConfig)
+
+  private val dummyManagerFactoryParameters = new ManagerFactoryParameters {}
+
+  "NettySSLHandler" when {
+    "no CryptoServiceProviderSetup" should {
+      "succeed" in {
+        createSSLSettingsAndCreateContext(None)
+
+        forAll(Constants.properties) {
+          System.getProperty(_) shouldBe null
+        }
+      }
+    }
+
+    "KeyManagerFactorySetup included" should {
+      "use given Provider without ManagerFactoryParameters" in {
+        createSSLSettingsAndCreateContext(Some(KeyManagerFactorySetup(TestProvider, None)))
+
+        System.getProperty(Constants.keyManagerFactoryInitWithKeystore) shouldBe Constants.accessed
+        System.getProperty(Constants.keyManagerFactoryInitWithSpec) shouldBe null
+
+        System.getProperty(Constants.trustManagerFactoryInitWithKeystore) shouldBe null
+        System.getProperty(Constants.trustManagerFactoryInitWithSpec) shouldBe null
+      }
+      "use given Provider with ManagerFactoryParameters" in {
+        createSSLSettingsAndCreateContext(Some(KeyManagerFactorySetup(TestProvider, Some(dummyManagerFactoryParameters))))
+
+        System.getProperty(Constants.keyManagerFactoryInitWithKeystore) shouldBe null
+        System.getProperty(Constants.keyManagerFactoryInitWithSpec) shouldBe Constants.accessed
+
+        System.getProperty(Constants.trustManagerFactoryInitWithKeystore) shouldBe null
+        System.getProperty(Constants.trustManagerFactoryInitWithSpec) shouldBe null
+      }
+    }
+    "TrustManagerFactorySetup included" should {
+      "use given Provider without ManagerFactoryParameters" in {
+        createSSLSettingsAndCreateContext(Some(TrustManagerFactorySetup(TestProvider, None)))
+
+        System.getProperty(Constants.keyManagerFactoryInitWithKeystore) shouldBe null
+        System.getProperty(Constants.keyManagerFactoryInitWithSpec) shouldBe null
+
+        System.getProperty(Constants.trustManagerFactoryInitWithKeystore) shouldBe Constants.accessed
+        System.getProperty(Constants.trustManagerFactoryInitWithSpec) shouldBe null
+      }
+      "use given Provider with ManagerFactoryParameters" in {
+        createSSLSettingsAndCreateContext(Some(TrustManagerFactorySetup(TestProvider, Some(dummyManagerFactoryParameters))))
+
+        System.getProperty(Constants.keyManagerFactoryInitWithKeystore) shouldBe null
+        System.getProperty(Constants.keyManagerFactoryInitWithSpec) shouldBe null
+
+        System.getProperty(Constants.trustManagerFactoryInitWithKeystore) shouldBe null
+        System.getProperty(Constants.trustManagerFactoryInitWithSpec) shouldBe Constants.accessed
+      }
+    }
+  }
+
+  private def createSSLSettingsAndCreateContext(setup: Option[CryptoServiceProviderSetup]) = {
+    val sslSettings = new SSLSettings(
+      knownFilesConfig,
+      setup.map(ActorSystemSetup.empty.and).getOrElse(ActorSystemSetup.empty)
```

Comment:
This is a bit weird, could you make the `setup` param a `Setup*` instead and just to `ActorSystemSetup(setup: _*)` instead?